File: src/backend/cs/StudyRoomSystem/StudyRoomSystem.Core/Structs/Entity/Room.cs
using System.ComponentModel.DataAnnotations;

namespace StudyRoomSystem.Core.Structs.Entity;

public class Room
{
    public Guid Id { get; set; } = Ulid.NewUlid().ToGuid();
    
    [MaxLength(64)]
    public required string Name { get; set; }
    
    public required TimeOnly OpenTime { get; set; }
    
    public required TimeOnly CloseTime { get; set; }
    
    [Range(1,2048)]
    public required int Rows { get; set; }
    
    [Range(1,2048)]
    public required int Cols { get; set; }

    public virtual ICollection<Seat> Seats { get; set; } = [];
}

File: src/frontend/web/study-room-system/src/lib/types/Room.ts
import type { Seat } from "./Seat";

export interface Room {
  id: string;
  name: string;
  openTime : string;
  closeTime : string;
  rows: number;
  cols: number;
  seats: Seat[] | null;
}

export interface RoomEdit {
  id: string;
  name: string;
  rows: number;
  cols: number;
  openHour: number;
  openMin: number;
  closeHour: number;
  closeMin: number;
}
